openlibrary/catalog/add_book/__init__.py

```diff
@@ -755,104 +1101,24 @@ def load(rec, account_key=None):
-
-    # Add cover to work, if needed
-    if not w.get('covers') and e.get_covers():
-        w['covers'] = [e['covers'][0]]
-        need_work_save = True
-
-    # Add description to work, if needed
-    if not w.get('description') and e.get('description'):
-        w['description'] = e['description']
-        need_work_save = True
-
-    # Add authors to work, if needed
-    if not w.get('authors'):
-        authors = [import_author(a) for a in rec.get('authors', [])]
-        w['authors'] = [
-            {'type': {'key': '/type/author_role'}, 'author': a.key}
-            for a in authors
-            if a.get('key')
-        ]
-        if w.get('authors'):
-            need_work_save = True
-
-    # Add ocaid to edition (str), if needed
-    if 'ocaid' in rec and not e.ocaid:
-        e['ocaid'] = rec['ocaid']
-        need_edition_save = True
-
-    # Add list fields to edition as needed
-    edition_list_fields = [
-        'local_id',
-        'lccn',
-        'lc_classifications',
-        'oclc_numbers',
-        'source_records',
-        'publishers',
-    ]
-    for f in edition_list_fields:
-        if f not in rec or not rec[f]:
-            continue
-        # ensure values is a list
-        values = rec[f] if isinstance(rec[f], list) else [rec[f]]
-        if f in e:
-            # get values from rec that are not currently on the edition
-            to_add = [v for v in values if v not in e[f]]
-            e[f] += to_add
-        else:
-            e[f] = to_add = values
-        if to_add:
-            need_edition_save = True
-
-    other_edition_fields = [
-        'number_of_pages',
-        'publish_date',
-    ]
-    for f in other_edition_fields:
-        if f not in rec or not rec[f]:
-            continue
-        if f not in e:
-            e[f] = rec[f]
-            need_edition_save = True
+    # Update edition fields from the record.
+    e, need_matched_edition_save = update_matched_edition(
+        e, rec, account_key, from_marc_record
+    )
 
-    # Add new identifiers
-    if 'identifiers' in rec:
-        identifiers = defaultdict(list, e.dict().get('identifiers', {}))
-        for k, vals in rec['identifiers'].items():
-            identifiers[k].extend(vals)
-            identifiers[k] = list(set(identifiers[k]))
-        if e.dict().get('identifiers') != identifiers:
-            e['identifiers'] = identifiers
-            need_edition_save = True
+    w = cast("Work", w)  # w is a work now rather than None.
+    w, need_matched_work_save = update_matched_work(w, e, rec)
 
     edits = []
-    reply = {
+    reply: dict = {
         'success': True,
         'edition': {'key': match, 'status': 'matched'},
         'work': {'key': w['key'], 'status': 'matched'},
     }
-    if need_edition_save:
+    if need_edition_save or need_matched_edition_save:
         reply['edition']['status'] = 'modified'
         edits.append(e.dict())
-    if need_work_save:
+    if need_work_save or need_matched_work_save:
```

Comment:
Can remove `need_work_save`

```suggestion
    if work_created or matched_work_modified:
```